doc/sfp-cmis/CMIS-custom-SI-settings.md

```diff
@@ -0,0 +1,256 @@
+      “PORT_MEDIA_SETTINGS” : {} 
+  },
+  
+  “RX_SETTING”: { 
+      “EQ_FIXED”: “true”, 
+      “GLOBAL_MEDIA_SETTINGS” : { 
+          “1-20”: { 
+              “100G_SPEED”: { 
+                  “CREDO”: { 
+                      “CREDO_PN”: { 
+                          “SI_PARAM_RX_OUTPUT_PRE” : { 
+                              “lane0” : “5”, 
+                              “lane1” : “5”,
+                              “lane2” : “5”,
+                              “lane3” : “5”,
+                              “lane4” : “5”,
+                              “lane5” : “5”,
+                              “lane6” : “5”,
+                              “lane7” : “5”,
+                          } 
+                      } 
+                  }, 
+                  “INNOLIGHT”: { 
+                      “INNOLIGHT_PN”: { 
+                          “SI_PARAM_RX_OUTPUT_POST” : { 
+                              “lane0” : “8”,
+                              “lane1” : “8”,
+                              “lane2” : “8”,
+                              “lane3” : “8”,
+                              “lane4” : “8”,
+                              “lane5” : “8”,
+                              “lane6” : “8”, 
+                              “lane7” : “8”, 
+                          } 
+                      } 
+                  } 
+              } 
+          }, 
+          “25,28,30”: { 
+              “100G_SPEED”: { 
+                  “Default”: { 
+                      “SI_PARAM_RX_INPUT_AMP” : { 
+                          “lane0” : “7”,
+                          “lane1” : “7”,
+                          “lane2” : “7”,
+                          “lane3” : “7”,
+                          “lane4” : “7”,
+                          “lane5” : “7”,
+                          “lane6” : “7”,
+                          “lane7” : “7” 
+                      } 
+                  } 
+              } 
+          } 
+      }, 
+      “PORT_MEDIA_SETTINGS” : { 
+          “32”: { 
+              “EOPTOLINK”: {
+                  "Default": {
+                      “SI_PARAM_RX_OUTPUT_PRE” : { 
+                          “lane0” : “9”,
+                          “lane1” : “9”,
+                          “lane2” : “9”,
+                          “lane3” : “9”,
+                          “lane4” : “9”,
+                          “lane5” : “9”,
+                          “lane6” : “9”,
+                          “lane7” : “9”,
+                      } 
+                  } 
+              } 
+          } 
+      }
+  }
+```
+
+# Proposed Work-Flow
+Please refer below points in line with flow diagram.
+
+1. When CMIS-supported module insertion happens in XCVRD, the module will progress to the AP_CONFIG state (after DP_DEINIT state) in the CMIS state machine. During which, when the module is in DataPathDeactivated or Disabled state, check if the optics_si_setting.json file is parsed successfully and if lane speed needs special Signal Integrity (SI) settings.
```

Comment:
What is the need of new file optics_si_setting.json?. We have media_settings.json file per platform basis which already has global and port media settings. If new attributes are needed, can we define it in media_settings.json instead. Could you please clarify it?.

Reply:
Yes. The optics_si_setting.json file template is based on media_settings.json.  
The media_settings.json file contains SAI attributes that are used by ASIC, whereas the optics_si_setting.json file contains OPTICS module SI setting attributes that are directly programmed on the modules.
The attributes are totally different for the media_settings and the optics__si_settings. 

Furthermore, it is not necessary for media settings to be required when optics_si_settings are applied. Both or Only one of them can be applicable, or none can be applicable.